scalyr_agent/monitor_utils/k8s.py

```diff
@@ -2183,14 +2197,25 @@ def query_api(self, path):
             # We suppress warnings here to avoid spam about an unverified connection going to stderr.
             # This method of warning suppression is not thread safe and has a small chance of suppressing warnings from
             # other threads if they are emitted while this request is going.
-            with warnings.catch_warnings():
-                warnings.simplefilter("ignore", category=Warning)
-                response = self._session.get(url, timeout=self._timeout, verify=False)
-            if self._kubelet_url.startswith("https://"):
-                global_log.warn(
-                    "Accessing Kubelet with an unverified HTTPS request.",
-                    limit_once_per_x_secs=3600,
-                    limit_key="unverified-kubelet-request",
+            verify_connection = self._verify_connection()
+            if not verify_connection:
+                with warnings.catch_warnings():
+                    warnings.simplefilter(
+                        "ignore",
+                        category=requests.packages.urllib3.exceptions.InsecureRequestWarning,
+                    )
+                    response = self._session.get(
```

Comment:
Let's create a method on this object called `_get(url, verify)` which simply invokes `_session.get`.  Use that method here and below.  Then, for testing, just do a mock path on the object to replace `_get` with a mock.  You can use that to write test cases for both checking `_get` is being invoked with the right parameters, and that the object fallsback appropriately when `_get` returns an error.